CustomerAPI/Controllers/CustomerController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Data.Entities;
+using Data.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CustomerAPI.Controllers
+{
+    [Route("api/controller")]
+    [ApiController]
+    public class CustomerController : ControllerBase
+    {
+        private ICustomerService _service;
+
+        public CustomerController(ICustomerService service)
+        {
+            _service = service ?? throw new ArgumentNullException(nameof(service));
+        }
+
+        [HttpPost]
+        public IActionResult Create(Customer customer)
+        {
+            try
+            {
+                _service.Create(customer);
+                return Created("Id: ", customer.Id);
+            }
+            catch (ArgumentException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+        }
+
+        [HttpGet("{Id}")]
+        public IActionResult GetById(long Id)
+        {
+            try
+            {
+                var customer = _service.GetById(Id);
+                return Ok(customer);
+            }
+            catch (ArgumentNullException exception)
+            {
+                return NotFound(exception.Message);
+            }
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            try
+            {
+                var listCustomers = _service.GetAll();
+                return Ok(listCustomers);
+            }
+            catch (Exception exception)
+            {
+                var exceptionMessage = exception.InnerException?.Message ?? exception.Message;
+                return Problem(exceptionMessage);
+            }
+        }
+
+        [HttpPut]
+        public IActionResult Update(Customer customer)
+        {
+            try
+            {
+                _service.Update(customer);
+                return Ok();
+            }
+            catch (ArgumentNullException exception)
+            {
+                return NotFound(exception.Message);
+            }
+            catch (ArgumentException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+        }
+
+        [HttpDelete]
+        public IActionResult Delete(long Id)
+        {
+            try
+            {
+                _service.Delete(Id);
+                return NoContent();
+            }
+            catch
+            {
+                return NotFound();
+            }
```

Comment:
Está estourando exceção mas em momento algum faz uso dela
![image](https://user-images.githubusercontent.com/96585621/203381629-0fb0e80f-5125-4188-9746-c8c5daf9bfae.png)
```suggestion
            catch (ArgumentException ex)
            {
                return NotFound(ex.Message);
            }
```